homeassistant/components/min_max/sensor.py

```diff
@@ -290,61 +288,80 @@ def _async_min_max_sensor_state_listener(
         self, event: EventType, update_state: bool = True
     ) -> None:
         """Handle the sensor state changes."""
-        new_state: State | None = event.data.get("new_state")
-        entity: str = event.data["entity_id"]
-
-        if (
-            new_state is None
-            or new_state.state is None
-            or new_state.state
-            in [
-                STATE_UNKNOWN,
-                STATE_UNAVAILABLE,
-            ]
-        ):
-            self.states[entity] = STATE_UNKNOWN
-            if not update_state:
-                return
+        calc_states: dict[str, float | str] = {}
 
-            self._calc_values()
-            self.async_write_ha_state()
-            return
+        entity_state: State | None = event.data.get("new_state")
+        entity: str = event.data["entity_id"]
 
-        if self._unit_of_measurement is None:
-            self._unit_of_measurement = new_state.attributes.get(
+        states: list[tuple[str, State]] = [
+            (entity_id, state)
+            for entity_id in self._entity_ids
+            if (state := self.hass.states.get(entity_id))
+        ]
+        for entity_id, new_state in states:
```

Comment:
`new_state` doesn't match well after the refactoring
```suggestion
        for entity_id, state in states:
```